src/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1875,6 +1875,37 @@ public void TestLongProcessIsWorking()
             Assert.True(p.HasExited);
         }
 
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        [ActiveIssue(37054, TestPlatforms.OSX)]
+        [Fact]
+        public void LongProcessNamesAreSupported()
+        {
+            string commandName = "sleep";
+
+            if (!IsProgramInstalled(commandName))
+            {
+                return;
+            }
+
+            string longProcessName = "123456789012345678901234567890";
+            string sleepCommandPathFileName = Path.Combine(TestDirectory, longProcessName);
+            File.Copy(GetProgramPath(commandName), sleepCommandPathFileName);
+
+            using (Process px = Process.Start(sleepCommandPathFileName, "30"))
```

Comment:
Since we expect to kill this process immediatley we are done with it, it could wait essentially forever instead of 30 seconds. That would avoid the case where the machines are so overloaded or unlucky that we don't read all the processes before the sleep expires. Put another way, this 30 seconds does not affect the length of the test.